packages/list-reusable-blocks/src/components/import-form/index.js

```diff
@@ -11,52 +11,47 @@ import { Button, Notice } from '@wordpress/components';
  */
 import importReusableBlock from '../../utils/import';
 
-class ImportForm extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			isLoading: false,
-			error: null,
-			file: null,
-		};
+function ImportForm( { instanceId, onUpload } ) {
+	const inputId = 'list-reusable-blocks-import-form-' + instanceId;
 
-		this.isStillMounted = true;
-		this.onChangeFile = this.onChangeFile.bind( this );
-		this.onSubmit = this.onSubmit.bind( this );
-	}
+	let isStillMounted = true;
```

Comment:
Should this be a "ref" instead?

Reply:
Hi @youknowriad, could you please help me to understand why should we implement `ref` instead of the current way? 

For my understanding I created two code sandbox and it turns out both are behaving same. I would personally prefer to go this way because
- First, we're able to achieve the expected outcome.
- Second, `ref` should be avoided as much as possible as mentioned in the official [docs](https://reactjs.org/docs/refs-and-the-dom.html#when-to-use-refs)

> Avoid using refs for anything that can be done declaratively.
- Third, it will add some unnecessary complexity.

**Code Sandbox link**
- Implementation using `ref` - [link to the sandbox](https://codesandbox.io/s/quirky-meninsky-hxh06)
- Implementation without `ref` - [link to the sandbox](https://codesandbox.io/s/dazzling-https-m4xzc)

Would love to hear your thoughts! 😄

Reply:
It comes down to safety for me, consider the `onSubmit` handler as written today, it is using a new instance on each render, just like isStillMounted, but if tomorrow we decide to memoize it (useCallback or useMemo for instance) for performance reasons, then the `isStillMounted` that it's going to use is going to be the wrong one because it will use the reference that was created when the callback has been first memoized (meaning isStillMounted will never be false)

Reply:
I have made the required changes, please take a look at it.